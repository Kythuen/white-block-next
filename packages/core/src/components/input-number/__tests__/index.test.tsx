import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import InputNumber from '../input-number.vue'
// import { useInputNumber } from '../use'

describe('# InputNumber', () => {
  describe('## Base', () => {
    it('### default config', () => {
      const wrapper = mount({
        render() {
          return <InputNumber />
        }
      })
      // expect(wrapper.classes()).toEqual(['wb-input-number'])
      // expect(wrapper.props()).toEqual({})
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  // describe('## Slots', () => {})

  // describe('## Events', () => {})

  // describe('## Props', () => {})

  // describe('## Composables', () => {})
})
