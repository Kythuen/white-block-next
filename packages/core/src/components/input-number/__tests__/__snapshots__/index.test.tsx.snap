// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`# InputNumber > ## Base > ### default config 1`] = `
<div
  class="wb-input-number wb-input-number--base wb-input-number--start wb-input-number--inner wb-input-number--md"
  flex="~ inline"
  focus-within="outline outline-2 outline-$wb-color-primary"
  gap="$wb-input-number-gap"
  items="center"
  overflow="hidden"
  p="0"
  relative=""
  ring="1 inset $wb-input-number-border"
  rounded="$wb-input-number-radius"
  w="$wb-input-number-width"
>
  <!---->
  <!-- TODO: 无法输入 -->
  <input
    bg="$wb-input-number-background"
    flex="1"
    focus-visible=""
    h="$wb-input-number-height"
    maxlength="15"
    overflow="hidden"
    p="x-$wb-input-number-padding"
    placeholder=""
    ring=""
    rounded="$wb-input-number-radius"
    step="1"
    text="$wb-input-number-text"
  />
  <!---->
  <div
    bg="$wb-color-soft"
    class="wb-input-number-options"
    flex="~ none col"
    gap="1px"
    h="[calc(var(--wb-input-number-height)-3px)]"
    m="r-px"
    p="b-px"
  >
    <wb-button
      data-key="add"
      disabled="false"
      focus-visible="outline-0 bg-$wb-color-primary"
      h="[calc(50%)]"
      rounded="rt-$wb-input-number-radius"
      size="md"
      tabindex="-1"
      theme="default"
      w="full"
    >
      <i
        i="tdesign-chevron-up"
      />
    </wb-button>
    <wb-button
      data-key="sub"
      disabled="false"
      focus-visible="outline-0 bg-$wb-color-primary"
      h="[calc(50%)]"
      rounded="rb-$wb-input-number-radius"
      size="md"
      tabindex="-1"
      theme="default"
      w="full"
    >
      <i
        i="tdesign-chevron-down"
      />
    </wb-button>
  </div>
</div>
`;
